test/reduce_test.hpp

```diff
@@ -0,0 +1,949 @@
+
+        // replicate
+        auto res         = output;
+        auto res_indices = indices;
+
+        Workspace idxspace{};
+        idxspace.Write(indices.data);
+
+        Workspace wspace{};
+        wspace.Write(workspace.data);
+
+        const double alpha64 = alpha;
+        const double beta64  = beta;
+
+        const void* const alphaPtr = (std::is_same<T, double>::value)
+                                         ? static_cast<const void*>(&alpha64)
+                                         : static_cast<const void*>(&alpha);
+        const void* const betaPtr  = (std::is_same<T, double>::value)
+                                         ? static_cast<const void*>(&beta64)
+                                         : static_cast<const void*>(&beta);
+
+        if(wspace.size() > 0)
+        {
+            reduce.ReduceTensor(get_handle(),
+                                idxspace.ptr(),
+                                idxspace.size(),
+                                wspace.ptr(),
+                                wspace.size(),
+                                alphaPtr,
+                                input.desc,
+                                input_dev.get(),
+                                betaPtr,
+                                output.desc,
+                                output_dev.get());
+        }
+        else
+        {
+            reduce.ReduceTensor(get_handle(),
+                                idxspace.ptr(),
+                                idxspace.size(),
+                                nullptr,
+                                0,
+                                alphaPtr,
+                                input.desc,
+                                input_dev.get(),
+                                betaPtr,
+                                output.desc,
+                                output_dev.get());
+        };
+
+        res.data         = handle.Read<T>(output_dev, res.data.size());
+        res_indices.data = idxspace.Read<decltype(res_indices.data)>();
+
+        return (std::make_tuple(res, res_indices));
+    }
+
+    void fail(int) const
+    {
+        std::cout << "verify_reduce_with_indices failed" << std::endl;
+        std::cout << "Input Tensor"
+                  << " " << input.desc.ToString() << std::endl;
+    }
+};
+
+template <class T>
+struct verify_reduce_no_indices
+{
+    miopen::ReduceTensorDescriptor reduce;
+    tensor<T> input;
+    tensor<T> output;
+    tensor<T> workspace;
+    float alpha;
+    float beta;
+
+    miopenReduceTensorOp_t reduceOp;
+    miopenDataType_t compTypeVal;
+    miopenNanPropagation_t nanOpt;
+
+    verify_reduce_no_indices( // NOLINT (hicpp-member-init)
+        const miopen::ReduceTensorDescriptor& reduce_,
+        const tensor<T>& input_,
+        const tensor<T>& output_,
+        const tensor<T>& workspace_,
+        float alpha_,
+        float beta_)
+    {
+        reduce    = reduce_;
+        input     = input_;
```

Comment:
The class member initialization should put in the initialization list instead of the constructor body, otherwise the Hip-tidy will complain when running CI.

Reply:
It has `// NOLINT (hicpp-member-init)`...

Reply:
> It has `// NOLINT (hicpp-member-init)`...

Well, that does not work for some reason because the hip-tidy does complain in CI...